mne/epochs.py

```diff
@@ -1324,71 +1417,43 @@ def save(self, fname):
-        fid = start_file(fname)
-
-        start_block(fid, FIFF.FIFFB_MEAS)
-        write_id(fid, FIFF.FIFF_BLOCK_ID)
-        if self.info['meas_id'] is not None:
-            write_id(fid, FIFF.FIFF_PARENT_BLOCK_ID, self.info['meas_id'])
-
-        # Write measurement info
-        write_meas_info(fid, self.info)
+        split_size : string | int
+            Large raw files are automatically split into multiple pieces. This
+            parameter specifies the maximum size of each piece. If the
+            parameter is an integer, it specifies the size in Bytes. It is
+            also possible to pass a human-readable string, e.g., 100MB.
+            Note: Due to FIFF file limitations, the maximum split size is 2GB.
 
-        # One or more evoked data sets
-        start_block(fid, FIFF.FIFFB_PROCESSED_DATA)
-        start_block(fid, FIFF.FIFFB_EPOCHS)
+            .. versionadded:: 0.10.0
 
-        # write events out after getting data to ensure bad events are dropped
-        data = self.get_data()
-        start_block(fid, FIFF.FIFFB_MNE_EVENTS)
-        write_int(fid, FIFF.FIFF_MNE_EVENT_LIST, self.events.T)
-        mapping_ = ';'.join([k + ':' + str(v) for k, v in
-                             self.event_id.items()])
-        write_string(fid, FIFF.FIFF_DESCRIPTION, mapping_)
-        end_block(fid, FIFF.FIFFB_MNE_EVENTS)
-
-        # First and last sample
-        first = int(self.times[0] * self.info['sfreq'])
-        last = first + len(self.times) - 1
-        write_int(fid, FIFF.FIFF_FIRST_SAMPLE, first)
-        write_int(fid, FIFF.FIFF_LAST_SAMPLE, last)
-
-        # save baseline
-        if self.baseline is not None:
-            bmin, bmax = self.baseline
-            bmin = self.times[0] if bmin is None else bmin
-            bmax = self.times[-1] if bmax is None else bmax
-            write_float(fid, FIFF.FIFF_MNE_BASELINE_MIN, bmin)
-            write_float(fid, FIFF.FIFF_MNE_BASELINE_MAX, bmax)
-
-        # The epochs itself
-        decal = np.empty(self.info['nchan'])
-        for k in range(self.info['nchan']):
-            decal[k] = 1.0 / (self.info['chs'][k]['cal'] *
-                              self.info['chs'][k].get('scale', 1.0))
-
-        data *= decal[np.newaxis, :, np.newaxis]
-
-        write_float_matrix(fid, FIFF.FIFF_EPOCH, data)
-
-        # undo modifications to data
-        data /= decal[np.newaxis, :, np.newaxis]
+        Notes
+        -----
+        Bad epochs will be dropped before saving the epochs to disk.
+        """
+        check_fname(fname, 'epochs', ('-epo.fif', '-epo.fif.gz'))
 
-        write_string(fid, FIFF.FIFFB_MNE_EPOCHS_DROP_LOG,
-                     json.dumps(self.drop_log))
+        if isinstance(split_size, string_types):
+            exp = dict(MB=20, GB=30).get(split_size[-2:], None)
+            if exp is None:
+                raise ValueError('split_size has to end with either'
+                                 '"MB" or "GB"')
```

Comment:
We do this somewhere else too I think, can you refactor int o a little helper function that takes `split_size` and returns a number of bytes?